be/src/exec/vectorized/join_hash_map.tpp

```diff
@@ -45,6 +45,49 @@ Status JoinBuildFunc<PT>::construct_hash_table(RuntimeState* state, JoinHashTabl
     return Status::OK();
 }
 
+template <PrimitiveType PT>
+void DirectMappingJoinBuildFunc<PT>::prepare(RuntimeState* runtime, JoinHashTableItems* table_items) {
+    table_items->bucket_size =
+            (int64_t)(std::numeric_limits<CppType>::max()) - (int64_t)(std::numeric_limits<CppType>::min() + 1l);
```

Comment:
1L << std::numeric_limits<CppType>::digits is more clear

Reply:
Fixed, replace with RunTimeTypeLimits<PT>::max_value()/RunTimeTypeLimits<PT>::min_value()